### DISCUSSION TITLE
ESP32 - Wyze Plug Outdoor - Flashed Tasmota OTA But Settings/Updates Will Not Persist

### DISCUSSION THREAD

**Discussion by ccuono** [arendst/Tasmota#17802] [Q&A]
Hello!

I managed to figure out a way to flash the [Wyze Plug Outdoor](https://www.wyze.com/products/wyze-plug-outdoor) from stock firmware to the current [release](http://ota.tasmota.com/tasmota32/release/) tasmota32.bin file.

The device boots up fine after flashing OTA and I can connect to it and access it via the web interface, however, any/all settings I set do not persist after the device reboots. When I try to connect to my wireless network it does wind up successfully connecting to my network showing the expected IP I've assigned it based on it's MAC and it reboots, but then after reboot, it's back to the default state broadcasting the same SSID and accessible via 192.168.4.1. This happens after applying any settings - once a reboot happens everything is back to the default state. I've also tried to apply different firmwares but I just get an error if I try to upload a firmware and it will just say that it's applying the firmware and rebooting if I try another OTA link but it'll just reboot back to the default state again...

What I'm trying to figure out is whether:

1. There's a way to get things to work in this state, or if the device is essentially stuck unless I can do a flash via serial cable
2. I should have used  [tasmota32.factory.bin](http://ota.tasmota.com/tasmota32/release/tasmota32.factory.bin) instead of tasmota32.bin or should have flashed with a different image like how it was with many of the 8266 OTA methods?

The problem with point 1 above: what I've been encountering with this plug, which I just received two days ago, is that even if I solder to the pins that are exposed and pull down GPIO0, it does NOT enter programming mode -- I've tried everything I can possibly do and it will NOT enter programming mode and I can't flash anything to it. I would have just went with this route IF it would enter programming mode, but it just simply will not do it.

Also: I can share further details on how I've managed to get the OTA flash to happen, however, I think we're in a big cat and mouse game when it comes to OTA flashing where if I post any solution or backdoors to make that happen, they'll get plugged right up into the future... Also, as it is right now, it looks like I managed to flash the firmware in a way where it's going to remain in this static initial state... no settings will persist and I can not have it join any networks to be of any use so it's probably not a good idea on how I've managed to achieve this _yet_.

**Comment by barbudor** [comment]
Just to remind:
- tasmota32.bin is meant for OTA upgrade (from tasmota to a more recent Tasmota), not for initial flashing
- tasmota32.factory.bin is meant for initial serial flashing replacing the original firmware

I understand that you have found a way to OTA the device from the original firmware but that doesn't fall in any of the standard way supported. You may have ended up with a device badly flashed, where some elements already present in the flash are interpreted as settings which create the problem

Now the behavior that you are experiencing looks like Tasmota triggers the recovery mecanism which erase all settings if Button 1 is pressed for more than 40 seconds
A typical case for that is when you configure a bad template using Button when it should be Button_i (inverted) or vice-versa
When Tasmota is in the configuration mode (soft AP, IP 192.168.4.1) try to go to the console with http://192.168.4.1/cs
And use command `Reset 1` to insure that setttings are erase
Tasmota will reboot
Then go to http://192.168.4.1/tp which should present you an empty template where everything is None or User
From there you can try to reconfigure the Wifi on http://192.168.4.1

Let us know

**Reply by ccuono** [reply]
Thanks for this advice -- I just gave this a shot but no luck, it still keeps being stuck on default whenever I apply anything and restart...

I double checked the insides from when I tried to do a serial flash and everything looks clean, nothing is crossed or has anything bad going on...

I did go through the `Reset 1` command and saw that `tp` shows a mix of `User` and `None` and I can see that it's `Based on ESP32-DevKit (1)` with nothing else selectable. I went ahead and changed everything to `None` but even after that it'll say that it saved the changes and reboot, just to have it revert back to the way it was with mixed `User` and `None` selections for each of the GPIOs...

**Reply by barbudor** [reply]
If you can power it from a safe PSU (not from mains) and hook a serial terminal, with `seriallog 3` may be it will show you some usefull logs just before restarting like the cause of the reboot

**Reply by ccuono** [reply]
I should be able to try and get to doing this over the next few days -- the big problem I was hitting while I had this wired up to a couple of the serial devices I've tried (a generic usb ftdi and a raspberry pi) I couldn't get anything out from it or communicate at all with it -- could be that I'm doing a bad job soldering or they gimped the leads somehow -- I do know if I tried to boot it with GPIO0 pulled low it would seemingly just do nothing even with a strong 3.3v supply but it would boot normally if I left GPIO0 alone...

it might prove to give me more information with tasmota now on it so hopefully it'll be fruitful.

**Reply by ccuono** [reply]
I attempted to try to get some serial data out from this device, it will NOT cooperate... I get no data -- I did manage to hit the en0 pin low a few times and that seems to have helped with something since now I am able to:

1.  Connect to my network and access the interface further on my own network
2. Apply the appropriate configuration template
3. Perform a firmware upgrade with tasmota32.bin by using the /up interface and uploading the firmware from my machine (it seems to fail if I try to do it direct from the OTA server... it just dies and no longer responds to anything)

When I check the /in interface, I can see `Partition ota_0` and `Partition ota_1` swap back and forth (the asterisk at the end of each will change from one to the other) as I do the firmware update...

The big bummer is that I am still hitting an issue where if the plug loses power I lose all the settings -- it goes back to square one with that default state with broadcasting the AP, it also seems to randomly restart back into the default state at some point in time... I'm not sure what's going on with that either.

I'm wondering if there's anything else I can try to get this to work properly and survive power loss... basically now I'm in a state where I can get this to work, but the settings will remain volatile and will be lost on poweroff...

**Reply by ccuono** [reply]
I built a firmware with esphome and that works fine on these devices taking the OTA method discovered. Since I can't get serial comm to work I'm just going to use that instead at this point... unfortunately I do not have time to continue with this and esphome is working so I'm just going to keep with that at this point.

There's good reason to suspect my UART device is messed up, however, I did get some spoofing with DNS to get an OTA firmware to flash... what's so mysterious is why settings wouldn't persist via this method. At least there is another method to employ that's more locked in with the overall implementation, but it lacks the flexibility of what tasmota has to offer. _oh well_ -- I followed this thread to figure out how to get the OTA method to work: https://github.com/HclX/WyzeUpdater/issues/9#issuecomment-885030254
